crates/astria-conductor/src/validation.rs

```diff
@@ -0,0 +1,537 @@
+use std::{
+    collections::HashMap,
+    str::FromStr,
+};
+
+use astria_sequencer_relayer::{
+    data_availability::{
+        SequencerNamespaceData,
+        SignedNamespaceData,
+    },
+    keys::public_key_to_address,
+    sequencer_block::SequencerBlock,
+    types::{
+        Commit,
+        CommitSig,
+    },
+};
+use bech32::{
+    self,
+    ToBase32,
+    Variant,
+};
+use color_eyre::eyre::{
+    bail,
+    ensure,
+    Result,
+    WrapErr as _,
+};
+use ed25519_dalek::Verifier;
+use prost::Message;
+use tendermint::{
+    account::Id as AccountId,
+    block::{
+        parts,
+        CommitSig as TendermintCommitSig,
+        Height,
+        Id as BlockId,
+        Round,
+    },
+    chain::Id as ChainId,
+    crypto,
+    merkle,
+    vote::{
+        self,
+        CanonicalVote,
+    },
+    Hash,
+    Signature,
+    Time,
+};
+use tendermint_proto::types::CommitSig as RawCommitSig;
+use tracing::{
+    instrument,
+    warn,
+};
+
+use crate::tendermint::{
+    TendermintClient,
+    ValidatorSet,
+};
+
+pub struct BlockVerifier {
```

Comment:
Two notes:

1. please make this crate-private, e.g. `pub(crate)`, as well as it's currently `pub` methods.
2. please add a do comment about the purpose of this type.

Reply:
I can't make it crate-private because tests/reader.rs needs it to be public :/ will add comment

Reply:
Gah, ok. This is part of my future refactor then